pkg/controllers/csi/driver/volumes/app/publisher.go

```diff
@@ -270,45 +264,52 @@ func (publisher *AppVolumePublisher) ensureMountSteps(ctx context.Context, bindC
 }
 
 func (publisher *AppVolumePublisher) hasTooManyMountAttempts(ctx context.Context, bindCfg *csivolumes.BindConfig, volumeCfg *csivolumes.VolumeConfig) (bool, error) {
-	volume, err := publisher.loadVolume(ctx, volumeCfg.VolumeID)
-	if err != nil {
+	appMount, err := publisher.db.ReadAppMount(ctx, metadata.AppMount{VolumeMetaID: volumeCfg.VolumeID})
+	if err != nil && errors.Is(err, gorm.ErrRecordNotFound) {
+		appMount = createNewAppMount(bindCfg, volumeCfg)
+		publisher.db.CreateAppMount(ctx, appMount)
+	} else if err != nil {
 		return false, err
 	}
 
-	if volume == nil {
-		volume = createNewVolume(bindCfg, volumeCfg)
-	}
-
-	if volume.MountAttempts > bindCfg.MaxMountAttempts {
+	if int(appMount.MountAttempts) > bindCfg.MaxMountAttempts {
 		return true, nil
 	}
 
-	volume.MountAttempts += 1
+	appMount.MountAttempts += 1
 
-	return false, publisher.db.InsertVolume(ctx, volume)
+	return false, publisher.db.UpdateAppMount(ctx, appMount)
 }
 
 func (publisher *AppVolumePublisher) storeVolume(ctx context.Context, bindCfg *csivolumes.BindConfig, volumeCfg *csivolumes.VolumeConfig) error {
-	volume := createNewVolume(bindCfg, volumeCfg)
-	log.Info("inserting volume info", "ID", volume.VolumeID, "PodUID", volume.PodName, "Version", volume.Version, "TenantUUID", volume.TenantUUID)
+	newAppMount := createNewAppMount(bindCfg, volumeCfg)
+	log.Info("inserting AppMount", "appMount", newAppMount)
 
-	return publisher.db.InsertVolume(ctx, volume)
-}
+	// check if it currently exists
+	readAppMount, err := publisher.db.ReadAppMount(ctx, metadata.AppMount{VolumeMetaID: newAppMount.VolumeMetaID})
+
+	if err != nil && errors.Is(err, gorm.ErrRecordNotFound) {
+		return publisher.db.CreateAppMount(ctx, newAppMount)
+	}
 
-func (publisher *AppVolumePublisher) loadVolume(ctx context.Context, volumeID string) (*metadata.Volume, error) {
-	volume, err := publisher.db.GetVolume(ctx, volumeID)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	return volume, nil
+	return publisher.db.UpdateAppMount(ctx, readAppMount)
 }
 
-func createNewVolume(bindCfg *csivolumes.BindConfig, volumeCfg *csivolumes.VolumeConfig) *metadata.Volume {
-	version := bindCfg.Version
-	if bindCfg.ImageDigest != "" {
-		version = bindCfg.ImageDigest
+func createNewAppMount(bindCfg *csivolumes.BindConfig, volumeCfg *csivolumes.VolumeConfig) *metadata.AppMount {
+	pr := metadata.PathResolver{RootDir: dtcsi.DataPath}
+
+	appMount := metadata.AppMount{
```

Comment:
you can return here, redundunt appMount variable.

Reply:
Done, thanks!